pkg/providers/vsphere/template.go

```diff
@@ -287,6 +293,13 @@ func buildTemplateMapMD(
 		if len(clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.CACertContent) > 0 {
 			values["registryCACert"] = clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.CACertContent
 		}
+
+		if clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Authenticate {
```

Comment:
Is this going to need populating on every provider? It seems like an extremely hard to maintain approach if so - did we brainstorm any alternatives by chance?